usl_pipeline/usl_lib/usl_lib/transformers/raster_transformers.py

```diff
@@ -0,0 +1,73 @@
+from typing import Tuple
+
+import numpy
+import numpy.typing as npt
+from shapely import geometry  # import Point, Polygon
+import geopandas as gpd
+
+from usl_lib.shared import geo_data
+
+
+def fill_in_soil_classes_missing_values_from_nearest_polygons(
+    elevation_header: geo_data.ElevationHeader,
+    green_areas_raster: npt.NDArray[numpy.float64],
+    soil_classes_raster: npt.NDArray[numpy.float64],
+    soil_classes_polygon_masks: list[Tuple[geometry.Polygon, int]],
+) -> list[Tuple[geometry.Polygon, int]]:
+    """Generates 1-cell polygons with nearest-neighbor soil classes for missing cells.
+
+    Args:
+        elevation_header: Elevation header providing coordinate system and the study
+            area region.
+        green_areas_raster: Raster with 0/1 masks indicating the presence of green area
+            for each raster cell.
+        soil_classes_raster: Raster with integer soil classes assigned to each raster
+            cell (0 means missing soil class).
+        soil_classes_polygon_masks: List of tuples containing polygons with associated
+            soil classes.
+
+    Returns:
+        The list of tuples containing polygons with associated soil classes for the
+        cells in a raster with present green area but missing soil class.
+    """
+    if len(soil_classes_polygon_masks) == 0:
```

Comment:
Yeah, we do need it since geopandas.GeoDataFrame.distance() fails if there is no geometry in the data frame. It might be legitimate case for some chunks even for soil classes data defined for the whole study area.